sonar-dotnet-shared-library/src/main/java/org/sonarsource/dotnet/shared/plugins/DotNetSensor.java

```diff
@@ -88,23 +88,33 @@ private boolean shouldExecuteOnProject(FileSystem fs) {
   private void executeInternal(SensorContext context) {
     UnaryOperator<String> toRealPath = new RealPathProvider();
 
+    boolean shouldSuggestScannerForMSBuild = false;
+
     List<Path> protobufPaths = reportPathCollector.protobufDirs();
     if (protobufPaths.isEmpty()) {
-      // We should consider failing fast in this case, see https://github.com/SonarSource/sonar-dotnet/issues/3603
-      LOG.warn("No protobuf reports found - no metrics and highlighting will be imported.");
+      LOG.warn("No protobuf reports found. The {} files will not have highlighting and metrics.", pluginMetadata.shortLanguageName());
+      shouldSuggestScannerForMSBuild = true;
     } else {
       protobufDataImporter.importResults(context, protobufPaths, toRealPath);
     }
 
     List<RoslynReport> roslynReports = reportPathCollector.roslynReports();
     if (roslynReports.isEmpty()) {
-      throw new IllegalStateException("No Roslyn issue reports were found.");
+      LOG.warn("No Roslyn issue reports were found. The {} files have not been analyzed.", pluginMetadata.shortLanguageName());
+      shouldSuggestScannerForMSBuild = true;
+    } else {
+      Map<String, List<RuleKey>> activeRoslynRulesByPartialRepoKey = activeRoslynRulesByPartialRepoKey(pluginMetadata, context.activeRules()
+        .findAll()
+        .stream()
+        .map(ActiveRule::ruleKey)
+        .collect(toList()));
+      roslynDataImporter.importRoslynReports(roslynReports, context, activeRoslynRulesByPartialRepoKey, toRealPath);
+    }
+
+    if (shouldSuggestScannerForMSBuild) {
+      LOG.warn("Your project contains {} files which cannot be analyzed with the scanner you are using."
+        + " To analyze C# or VB.NET files, you must use the Scanner for MSBuild 4.x, see https://redirect.sonarsource.com/doc/install-configure-scanner-msbuild.html",
```

Comment:
_Optional:_ I'm thinking if `C# or VB.NET` files is exact enough. We'll have more types like `.razor` in the future.

```suggestion
        + " To analyze .NET project files, you must use the Scanner for MSBuild 4.x, see https://redirect.sonarsource.com/doc/install-configure-scanner-msbuild.html",
```

Reply:
Good point, I'll change to `C# or VB.NET code` - which could be in razor files as well

.NET project files can be analyzed by the `scanner-cli` - TS, JS, HTML (as in "referenced in a .NET project)